### PR TITLE
sched: minor time slicing cleanup

### PR DESCRIPTION
**Pull request by npitre**
- Make `sliceable()` the actual condition for a sliceable thread.
- Avoid creating a slice timeout for non sliceable threads.
- Always reset `slice_expired` even if the next thread is not sliceable.
- Fold `slice_expired_locked()` into `z_time_slice()` to avoid the hidden
  unlock/lock sequence.
- Change `curr` to `thread` as this is not necessarily the current
  thread (yet) being set.
- Make variables static.

### CODE REVIEW

File: kernel/sched.c [hunk 1]
```diff
@@ -492,63 +507,33 @@ void k_thread_time_slice_set(struct k_thread *th, int32_t slice_ticks,
...
-	if (slice_time(_current) && sliceable(_current)) {
-		if (slice_expired[_current_cpu->id]) {
-			/* Note: this will (if so enabled) internally
-			 * drop and reacquire the scheduler lock
-			 * around the callback!  Don't put anything
-			 * after this line that requires
-			 * synchronization.
-			 */
-			key = slice_expired_locked(key);
+	if (sliceable(curr) && slice_expired[_current_cpu->id]) {
```
Comment (andyross) [inline kernel/sched.c:525]: This bit seems like a mild performance regression for the (common) non-timeslicing case: the previous version would short circuit rapidly if the current thread was not configured to slice (which is just checking a global variable for zero, also the thread struct equivalent if PER_THREAD).  Now it needs to go down the list of conditions in sliceable(), which is a more involved test.

File: kernel/sched.c [hunk 2]
```diff
@@ -414,30 +414,45 @@ static void flag_ipi(void)
 
 static int slice_ticks;
 static int slice_max_prio;
-struct _timeout slice_timeouts[CONFIG_MP_MAX_NUM_CPUS];
-bool slice_expired[CONFIG_MP_MAX_NUM_CPUS];
+static struct _timeout slice_timeouts[CONFIG_MP_MAX_NUM_CPUS];
+static bool slice_expired[CONFIG_MP_MAX_NUM_CPUS];
```
Comment (andyross) [inline kernel/sched.c:418]: Oops.  I do still think that these naturally ought to be part of the CPU struct.

**Review by andyross — COMMENTED** [review]
One performance note.  On the whole it all looks great.

**Comment by npitre** [comment]
- made the test on expired condition first in `z_time_slice()`
- reordered conditions in `sliceable()`so the most likely to be false are first

**Review by andyross — APPROVED** [review]
looks great

**Review by ceolin — APPROVED** [review]

**Pull request merged**